cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/PythonAddDeclarationsPass.kt

```diff
@@ -155,12 +155,28 @@ class PythonAddDeclarationsPass(ctx: TranslationContext) : ComponentPass(ctx), L
         // If we didn't create any field up to this point and if we are still have not returned, we
         // can create a normal variable. We need to take scope modifications into account.
         var decl =
-            if (field == null && targetScope != null) {
-                scopeManager.withScope(targetScope) { newVariableDeclaration(ref.name) }
-            } else if (field == null) {
-                newVariableDeclaration(ref.name)
-            } else {
-                field
+            when {
```

Comment:
This is not a good solution to the problem. Actually the problem lies elsewhere. The code in the original method is correct, but the scoping is "wrong".

`refA` in your example case lives in a "Function Scope", where as the current scope at the time of the resolution is a "Local Scope". Therefore, the lookup in line 115 is not successful. Actually both should be in the same scope since Python does not have any local scopes, only function scopes. So we could either try to set the correct scope (which is probably not that easy), or we need to adapt the scope filter in line 115. But I am strongly against this solution here that introduces two additional lookups that are unnecessary.